### DISCUSSION TITLE
Chapter 1 Task 9 question

### DISCUSSION THREAD

**Discussion by RazzorLabs** [kowainik/learn4haskell#510] [General]
Hi,

So I am trying to figure out how to solve this. I am able to sum all digits:

`sumLastTwo 0 = 0
sumLastTwo x = (x `mod` 10) + sumLastTwo (x `div` 10)`

With `mod` I can get last digit, but how to take number before the last one? 
I have tried to break digit in to the characters but no idea how to work with output afterwards.

**Comment by RazzorLabs** [comment]
I think I got it:

`sumLastDigits x =
   let lastDigit = mod x 10
       lastTwoDigits = mod x 100
   in lastDigit + (lastTwoDigits `div` 10)`

**Comment by chshersh** [comment]
@RazzorLabs Yes, that looks like a correct solution 🙂 
Happy to see that you've figured it out! 👏🏻

**Comment by RazzorLabs** [comment]
I am fighting with Task 10 in Chapter 1. I have came with one idea how to do it. So I would like to split number on digits with:

`splitToDigits n 
            | n < 1 = []
            | otherwise = splitToDigits (div n 10) ++ [mod n 10]`

After that I would like to apply "head" on the list of char which should be working:

*Main> head [1, 4, 5]
1
*Main>

but I have no idea how to filter the ouptut using the head in the function which I have created. I know this is probably not the best solution but how can I do it this way?

thx

**Comment by chshersh** [comment]
@RazzorLabs I'm not sure I understand your question. But if you write a separate function `splitDigits` with the type `Integer -> [Integer]` as you described, you should be able to call `head` on the result of this function, i.e.

```haskell
firstDigit :: Integer -> Integer
firstDigit n = head (splitDigits n)
```

**Reply by RazzorLabs** [reply]
Ouch. Nice. This is exactly what I was looking for. Thank you very much for the help. In the meantime I have found another way how to solve this. btw, thank you for this course, it is really amazing :)
